git-sync/main.go

```diff
@@ -168,12 +184,26 @@ func syncRepo(repo, dest, branch, rev string, depth int) error {
 
 	if *flChmod != 0 {
 		// set file permissions
-		_, err = runCommand("chmod", "", []string{"-R", string(*flChmod), dest})
+		_, err = runCommand("chmod", "", []string{"-R", string(*flChmod), pullTarget})
 		if err != nil {
 			return err
 		}
 	}
 
+	if atomic && strings.Compare(pullTarget, path.Join(dest, subdir)) != 0 {
+		args := []string{"-snf", pullTarget, "data-link"}
+		_, err := runCommand("ln", dest, args)
+		if err != nil {
+			return fmt.Errorf("error creating symbolic link: %v", err)
+		}
+
```

Comment:
Add a comment explaining why we do an `ln` followed by an `mv` (and not just a single `ln`)?